Lesson4/src/com/home/RecruitingOffice.java

```diff
@@ -0,0 +1,40 @@
+package com.home;
+
+
+import com.home.model.Address;
+import com.home.model.Person;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class RecruitingOffice {
+    private final Person[] recruits;
+
+    public RecruitingOffice(PersonRegistry personRegistry, Address address) {
+        List<Person> registeredPersons = personRegistry.getPeople(address);
+        List<Person> tmpListPersons = new LinkedList<>();
+        // Checking of requirements
+        for (Person person : registeredPersons) {
+            if (person.getAge() < 27 && person.getAge() > 18 && person.getGender() == 'M') {
+                tmpListPersons.add(person);
+            }
+        }
+        // List to Array
+        Person[] tmpArrPersons = new Person[tmpListPersons.size()];
+        for (int i = 0; i < tmpArrPersons.length; i++) {
+            tmpArrPersons[i] = tmpListPersons.get(i);
+        }
+
+        this.recruits = tmpArrPersons;
```

Comment:
I'd suggest not to store recruits, but to store personRegistry

In that way you would be able to determine military fit people at every moment (because new people can be registered in the personRegistry)

So what i'd suggest here is just to save PersonRegistry in constructor and creating a separate method to calculate and show recruits

Reply:
Now I understand, thx.